src/ngraph/runtime/cpu/op/conv_add.cpp

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+* Copyright 2017-2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <numeric>
+
+#include "conv_add.hpp"
+
+#include "ngraph/op/convolution.hpp"
+#include "ngraph/op/get_output_element.hpp"
+#include "ngraph/util.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+void op::util::validate_conv_shapes(const Shape& data_shape, const Shape& filters_shape)
+{
+    if (data_shape[1] != filters_shape[1])
+    {
+        throw ngraph_error(
```

Comment:
please switch to NGRAPH_TYPE_CHECK; see other ops for an example

Reply:
Strangely, I couldn't find any examples for NGRAPH_TYPE_CHECK, even on master.
Can you please point me to a specific example?

Reply:
I think @Krovatkin is referring to NODE_VALIDATION_ASSERT

Reply:
Great! Thanks. Now there are lot of examples.
will do the change